src/schema/v2/me/myCollectionDeleteArtworkMutation.ts

```diff
@@ -22,16 +22,13 @@ export const myCollectionDeleteArtworkMutation = mutationWithClientMutationId<
       resolve: (result) => result,
     },
   },
-  mutateAndGetPayload: async (
-    { artworkId },
-    { myCollectionDeleteArtworkLoader }
-  ) => {
-    if (!myCollectionDeleteArtworkLoader) {
+  mutateAndGetPayload: async ({ artworkId }, { deleteArtworkLoader }) => {
+    if (!deleteArtworkLoader) {
       return new Error("You need to be signed in to perform this action")
     }
 
     try {
-      const response = await myCollectionDeleteArtworkLoader(artworkId)
+      const response = await deleteArtworkLoader(artworkId, { permanent: true })
```

Comment:
It looks like you're dropping in `deleteArtworkLoader` for the collection-specific one, but wouldn't they resolve differently?

ie- the artwork delete loader calls an endpoint which returns [an artwork](https://github.com/artsy/gravity/blob/036472976c2b777827b0560c24f4a5171db7c84b/app/api/v1/artworks_endpoint.rb#L120), whereas the collection artwork delete loader calls an endpoint which returns [the collection](https://github.com/artsy/gravity/blob/036472976c2b777827b0560c24f4a5171db7c84b/app/api/v1/collections_artworks_endpoint.rb#L88)

Reply:
@mzikherman true and good catch! @damassi can you comment on this change in contract with Eigen? Will this cause any issues? I tried this locally and didn't see anything but things are moving fast and I might have missed something.

Reply:
As I understand it, an artwork is an artwork as long as there's a valid artwork ID, and so using the root delete artwork loader makes since assuming the callbacks are in place in Gravity to properly remove that artwork from a collection a users collection per https://github.com/artsy/gravity/pull/13552.

Reply:
No I mean there's an issue in Metaphysics (and possibly clients?). The shape of the `response` is now different I imagine (since it's an artwork JSON vs. a collection JSON), and so would there be some code in this mutation resolver, where the `MutationSuccess` might fail?

like:
```
collectionDeleteArtworkMutation(artworkID: ...) {
   ... on MutationSuccess {
    # Are these fields real, or cause a crash?
    collection {
      id
    }
    artwork {
      id
    }
  }
}
```

Reply:
Basically the `const response = ...` is now a different shape, so it seems odd to be able to drop it in 1 for 1, w/o some other tweaks further down in this file to things that resolve or operate on `response`?

Reply:
The way that we're doing it right now in Eigen is if the artwork deletion operation was a success, it returns `success` -- and we then use the artwork ID of the deleted artwork to find the stale item in the list and remove it: https://github.com/artsy/eigen/blob/60a223103b852c6b43e09d2f84570436a5b58831/src/lib/Scenes/MyCollection/State/MyCollectionArtworkModel.tsx#L466-L500

Does this answer things?

Reply:
Sure! But my original question still remains.

How would the success type get properly resolved? The `isTypeOf` expects that the response is a collection, and not an artwork: https://github.com/artsy/metaphysics/blob/bc00d1fa80346208feaafc78268e261c5db9d24f/src/schema/v2/me/myCollection.ts#L144

The spec (which still passes), needs to have it's stub updated as well: https://github.com/artsy/metaphysics/pull/2755/files#diff-4b7d51616bf5dd00ec99b223c0bda166R46

The field `name: "My Collection"` will no longer appear in the response from the loader.

Am I missing something?

Reply:
Ok I tried this again this morning and I'm getting errors when loading the list page after deleting the artwork. I took the nudge from @mzikherman and updated that line he's highlighted to look more like this:

```
return data.deleted
```

So if the response includes the deleted key and it's set to true (which it does), then that should return true and satisfy the contract. Except the list page blows up and I'm not quite sure where to head next. 🤔